wire-ios-sync-engine/Source/Synchronization/Strategies/UserClientEventConsumer.swift

```diff
@@ -25,38 +25,50 @@ import Foundation
 @objcMembers
 public class UserClientEventConsumer: NSObject, ZMEventAsyncConsumer {
 
-    let managedObjectContext: NSManagedObjectContext
-    let clientRegistrationStatus: ZMClientRegistrationStatus
-    let clientUpdateStatus: ClientUpdateStatus
+    private let managedObjectContext: NSManagedObjectContext
+    private let clientRegistrationStatus: ZMClientRegistrationStatus
+    private let clientUpdateStatus: ClientUpdateStatus
+    private let useCaseFactory: UseCaseFactoryProtocol
 
-    public init (managedObjectContext: NSManagedObjectContext,
-                 clientRegistrationStatus: ZMClientRegistrationStatus,
-                 clientUpdateStatus: ClientUpdateStatus) {
+    public init(
+        managedObjectContext: NSManagedObjectContext,
+        clientRegistrationStatus: ZMClientRegistrationStatus,
+        clientUpdateStatus: ClientUpdateStatus,
+        useCaseFactory: UseCaseFactoryProtocol
+    ) {
         self.managedObjectContext = managedObjectContext
         self.clientRegistrationStatus = clientRegistrationStatus
         self.clientUpdateStatus = clientUpdateStatus
-
+        self.useCaseFactory = useCaseFactory
         super.init()
     }
 
-    public func processEvents(_ events: [WireTransport.ZMUpdateEvent], liveEvents: Bool, prefetchResult: ZMFetchRequestBatchResult?) async {
+    public func processEvents(
+        _ events: [ZMUpdateEvent],
+        liveEvents: Bool,
+        prefetchResult: ZMFetchRequestBatchResult?
+    ) async {
         for event in events {
-            await processUpdateEvent(event)
+            do {
+                try await processUpdateEvent(event)
+            } catch {
+                WireLogger.updateEvent.error("failed to process user client event: \(event): \(error)")
```

Comment:
I would expose error to logfile on prod, what do you think something like:
```suggestion
                WireLogger.updateEvent.error("failed to process user client event: \(event.safeForLogging): \(error)", attributes: .safePublic)
```